### PR TITLE
Add menu to console wallet

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
- Added menu to console wallet
- Small user feedback improvement with transactions tab
- Clippy fixes

## Motivation and Context
Better user experience

## How Has This Been Tested?
Running the console wallet

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [X] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I ran `cargo test` successfully before submitting my PR.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_console_wallet/src/ui/app.rs [hunk 1]
```diff
@@ -126,6 +130,10 @@ impl<B: Backend> App<B> {
         self.tabs.on_down(&mut self.app_state);
     }
 
+    pub fn on_f10(&mut self) {
```
Comment (delta1) [inline applications/tari_console_wallet/src/ui/app.rs:133]:
```suggestion
    pub fn should_quit(&mut self) {
```

File: applications/tari_console_wallet/src/ui/components/menu.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            Span::styled(
+                "NextTab ",
+                Style::default()
+                    .fg(Color::Magenta)
+                    .bg(Color::LightGreen)
+                    .add_modifier(Modifier::BOLD),
+            ),
+        ]);
+        let quit = Spans::from(vec![
+            Span::styled("F10/Ctrl-Q/Ctrl-C", Style::default().fg(Color::Green)),
```
Comment (delta1) [inline applications/tari_console_wallet/src/ui/components/menu.rs:49]:
```suggestion
            Span::styled("F10/Ctrl-Q ", Style::default().fg(Color::Green)),
```

File: applications/tari_console_wallet/src/ui/mod.rs [hunk 3]
```diff
@@ -105,6 +105,7 @@ fn crossterm_loop(app: App<CrosstermBackend<Stdout>>) -> Result<(), ExitCodes> {
                 (KeyCode::Backspace, _) => app.on_backspace(),
                 (KeyCode::Enter, _) => app.on_key('\n'),
                 (KeyCode::Tab, _) => app.on_key('\t'),
+                (KeyCode::F(10), _) => app.on_f10(),
```
Comment (delta1) [inline applications/tari_console_wallet/src/ui/mod.rs:108]:
```suggestion
                (KeyCode::F(10), _) => app.should_quit(),
```

File: applications/tari_console_wallet/src/ui/components/menu.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            Span::styled(":", Style::default().fg(Color::White)),
+            Span::styled(
+                "PrevTab ",
+                Style::default()
+                    .fg(Color::Magenta)
+                    .bg(Color::LightGreen)
+                    .add_modifier(Modifier::BOLD),
+            ),
+            Span::raw(" "),
+            Span::styled("Tab/RightArrow", Style::default().fg(Color::Green)),
```
Comment (delta1) [inline applications/tari_console_wallet/src/ui/components/menu.rs:38]:
```suggestion
            Span::styled("Tab/RightArrow ", Style::default().fg(Color::Green)),
```

File: applications/tari_console_wallet/src/ui/components/menu.rs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+impl<B: Backend> Component<B> for Menu {
+    fn draw(&mut self, f: &mut Frame<B>, area: Rect, _app_state: &AppState)
+    where B: Backend {
+        let columns = Layout::default()
+            .direction(Direction::Horizontal)
+            .constraints([Constraint::Ratio(4, 5), Constraint::Ratio(1, 5)].as_ref())
+            .split(area);
+
+        let others = Spans::from(vec![
+            Span::styled("LeftArrow", Style::default().fg(Color::Green)),
```
Comment (delta1) [inline applications/tari_console_wallet/src/ui/components/menu.rs:28]:
```suggestion
            Span::styled(" LeftArrow", Style::default().fg(Color::Green)),
```

**Review by delta1 — CHANGES_REQUESTED** [review]
Nice work, some minor changes I would like but not the end of the world

File: applications/tari_console_wallet/src/ui/components/menu.rs (hunk 2 above)
Comment (hansieodendaal) [reply applications/tari_console_wallet/src/ui/components/menu.rs:49]: Please elaborate on why `Ctrl-C` should not be included/disclosed?

File: applications/tari_console_wallet/src/ui/components/menu.rs (hunk 4 above)
Comment (hansieodendaal) [reply applications/tari_console_wallet/src/ui/components/menu.rs:38]: The space at the end is a styling thing on the TUI. Maybe we should get a vote on it; either way can work for me.

File: applications/tari_console_wallet/src/ui/components/menu.rs (hunk 5 above)
Comment (hansieodendaal) [inline applications/tari_console_wallet/src/ui/components/menu.rs:28]: As above

**Comment by hansieodendaal** [comment]
> Nice work, some minor changes I would like but not the end of the world

Thanks

File: applications/tari_console_wallet/src/ui/components/menu.rs (hunk 2 above)
Comment (delta1) [reply applications/tari_console_wallet/src/ui/components/menu.rs:49]: Just personal preference really, I think 2 options is enough but all good

File: applications/tari_console_wallet/src/ui/components/menu.rs (hunk 4 above)
Comment (delta1) [reply applications/tari_console_wallet/src/ui/components/menu.rs:38]: Sorry I actually meant a space after the colon for both

image

File: applications/tari_console_wallet/src/ui/components/menu.rs (hunk 2 above)
Comment (StriderDM) [reply applications/tari_console_wallet/src/ui/components/menu.rs:49]: I think what @delta1 is saying here is that it is reasonable to assume that the user would think ctrl-c would terminate the application without it needing to be explicitly mentioned.

Either way should be fine though.

**Review by stringhandler — APPROVED** [review]

**Pull request merged**